src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -92,14 +92,20 @@ private IVariableDeclaratorOperation CreateVariableDeclarator(BoundLocal boundLo
             return new VariableDeclarator(boundLocal.LocalSymbol, initializer: null, semanticModel: _semanticModel, syntax: boundLocal.Syntax, type: null, constantValue: default, isImplicit: false);
         }
 
-        private IOperation CreateBoundCallInstanceOperation(BoundCall boundCall)
+        private IOperation CreateReceiverOperation(BoundNode instance, ISymbol symbol)
         {
-            if (boundCall.Method == null || boundCall.Method.IsStatic)
+            if (instance == null || instance.Kind == BoundKind.TypeExpression)
             {
                 return null;
             }
 
-            return Create(boundCall.ReceiverOpt);
+            // Static members cannot have an implicit this receiver
+            if (symbol != null && symbol.IsStatic && instance.WasCompilerGenerated && instance.Kind == BoundKind.ThisReference)
+            {
+                return null;
+            }
+
+            return Create(instance);
         }
```

Comment:
Consider having `CreateReceiverOperation` return a `Lazy<IOperation>` and return a singleton instance of `Lazy<IOperation>` in the static case. Same comment for VB.

Reply:
Done.